easybuild/tools/module_generator.py

```diff
@@ -238,18 +283,199 @@ def set_alias(self, key, value):
+        """
+        template = 'prepend_path(%s,%s)\n'
+
+        if isinstance(paths, basestring):
+            self.log.info("Wrapping %s into a list before using it to prepend path %s" % (paths, key))
+            paths = [paths]
+
+        # make sure only relative paths are passed
+        for i in xrange(len(paths)):
+            if os.path.isabs(paths[i]) and not allow_abs:
+                self.log.error("Absolute path %s passed to prepend_paths which only expects relative paths." % paths[i])
+            elif not os.path.isabs(paths[i]):
+                # prepend $root (= installdir) for relative paths
+                paths[i] = ' pathJoin(pkg.root,"%s")' % paths[i]
+
+        statements = [template % (quote_str(key), p) for p in paths]
+        return ''.join(statements)
+
+    def use(self, paths):
+        """
+        Generate module use statements for given list of module paths.
+        """
+        use_statements = []
+        for path in paths:
+            use_statements.append('use("%s")' % path)
+        return '\n'.join(use_statements)
+
+
+    def set_environment(self, key, value):
+
+        """
+        Generate setenv statement for the given key/value pair.
+        """
+        # quotes are needed, to ensure smooth working of EBDEVEL* modulefiles
+        return 'setenv("%s", %s)\n' % (key, quote_str(value))
+
+
+    def msg_on_load(self, msg):
+        """
+        Add a message that should be printed when loading the module.
+        """
+        pass
+
+
+    def add_tcl_footer(self, tcltxt):
+        """
+        Append whatever Tcl code you want to your modulefile
+        """
+    #@todo to pass or not to pass? this should fail in the context of generating Lua modules
+        pass
+
+
+    def set_alias(self, key, value):
+        """
+        Generate set-alias statement in modulefile for the given key/value pair.
+        """
+    # quotes are needed, to ensure smooth working of EBDEVEL* modulefiles
+        return 'setalias(%s,"%s")\n' % (key, quote_str(value))
+
+
+def avail_module_generators():
+    """
+    Return all known module syntaxes.
+    """
+    class_dict = {}
+    for klass in get_subclasses(ModuleGenerator):
+        class_name = klass.__name__
+        if class_name.startswith(MODULE_GENERATOR_CLASS_PREFIX):
+            syntax = class_name[len(MODULE_GENERATOR_CLASS_PREFIX):]
+            class_dict.update({syntax: klass})
+        else:
+            tup = (MODULE_GENERATOR_CLASS_PREFIX, class_name)
+            _log.error("Invalid name for ModuleGenerator subclass, should start with %s: %s" % tup)
+    return class_dict
+
+
+def module_generator(app, fake=False):
+    """
+    Return interface to modules tool (environment modules (C, Tcl), or Lmod)
+    """
+    module_syntax = get_module_syntax()
+    module_generator_class = avail_module_generators().get(module_syntax)
+    return module_generator_class(app, fake=fake)
+
+def return_module_loadregex(modfilepath):
```

Comment:
It's not because you're generating `.lua` module files in this run, that you'll only have `.lua` modules on the system. Lmod can consume both.

So you can't ask the current module generator how to parse `load`s from a module file.

Also, we have to be careful not to introduce a cyclic import situation again.